### PR TITLE
add Team tests

### LINKED ISSUE #297: Team entity tests
Write unit tests for Team class. Tests should evaluate all possible changes to the class and ensure valid state after each modification.

- [x] #324 Add new UTs

### PR DESCRIPTION
**Pull request by NataliiaBidiuk**
### Summary

add new file TeamTests_

close #297
#### Design

- [ ] Web API layer contains as little logic as possible
- [ ] Domain objects use semantic names

#### Perfromance

- [ ] In case of any DB query was changed: attach generated SQL for review
- [ ] Number of DB roundtrips should be as low as possible

#### Unit Tests

- [ ] Naming: Initial state and expected result are properly stated
- [ ] Test follows [AAA](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) pattern.
- [ ] Test is Isolated
- [ ] Mock instances and expected instances do not share references
- [ ] When new property is added all related comparers are updated

### CODE REVIEW

File: tests/VolleyManagement.Domain.UnitTests/TeamTests.cs [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        {
+            //Arrange
+            Team team = null;
+
+            //Act
+            team = CreateTeam();
+
+            //Assert
+            AssertCorrectTeamCreated(team);
+
```
Comment (sdiachen) [inline tests/VolleyManagement.Domain.UnitTests/TeamTests.cs]: ?

File: tests/VolleyManagement.Domain.UnitTests/TeamTests.cs [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        {
+            //Arrange
+            var team = CreateTeam();
+
+            //Act
+            Action act = () =>
+            {
+                team.Roster = new List<PlayerId>() {
+                new PlayerId() {
+                    Id = Constants.Team.MIN_ID - 1
```
Comment (sdiachen) [inline tests/VolleyManagement.Domain.UnitTests/TeamTests.cs]: Why PlayerId is not immutable?

**Comment by sdiachen** [comment]
Fix build errors

File: tests/VolleyManagement.Domain.UnitTests/TeamTests.cs (hunk 2 above)
Comment (Myroslavv) [reply tests/VolleyManagement.Domain.UnitTests/TeamTests.cs]: It is now.

Comment (Myroslavv) [reply tests/VolleyManagement.Domain.UnitTests/TeamTests.cs]: Which is why it wasn't changed previously.

**Pull request merged**